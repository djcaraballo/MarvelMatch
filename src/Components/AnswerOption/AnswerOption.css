@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

li {
  list-style-type: none;
  color: #fff;
  font-family: 'Orbitron', sans-serif;
  font-size: 1.5em;
  line-height: 1.5em;
}

.radio-button {
  margin: 10px;
  font-size: 2.5em;
  vertical-align: text-bottom;
}

.radio-button:hover {
  cursor: pointer;
}