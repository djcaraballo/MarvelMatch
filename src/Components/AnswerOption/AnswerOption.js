import React from 'react'
import PropTypes from 'prop-types'

const AnswerOption = (props) => {
  const { answerType, answerContent, answer, onAnswerSelected } = props
  return (
    <li>
      <input
        type='radio'
        className='radio-button'
        name='radio-group'
        checked={answerType === answer}
        id={answerType}
        value={answerType}
        disabled={answer}
        onChange={onAnswerSelected}
      />
      <label className='radio-label' htmlFor={answerType}>
        {answerContent}
      </label>
    </li>
  )
}

AnswerOption.propTypes = {
  answerType: PropTypes.string,
  answerContent: PropTypes.string,
  answer: PropTypes.string,
  onAnswerSelected: PropTypes.func,
}

export default AnswerOption

