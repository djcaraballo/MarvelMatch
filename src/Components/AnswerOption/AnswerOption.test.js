import React from 'react'
import AnswerOption from './AnswerOption'
import { shallow } from 'enzyme'

describe('AnswerOption Component', () => {
  let wrapper
  let mockAnswerType = ''
  let mockAnswerContent = ''
  let mockAnswer = ''
  let mockOnAnswerSelected = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <AnswerOption 
        answerType={mockAnswerType}
        answerContent={mockAnswerContent}
        answer={mockAnswer}
        onAnswerSelected={mockOnAnswerSelected}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})