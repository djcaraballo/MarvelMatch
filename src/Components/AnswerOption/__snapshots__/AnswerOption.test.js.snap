// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerOption Component should match the snapshot 1`] = `
<li>
  <input
    checked={true}
    className="radio-button"
    disabled=""
    id=""
    name="radio-group"
    onChange={[MockFunction]}
    type="radio"
    value=""
  />
  <label
    className="radio-label"
    htmlFor=""
  />
</li>
`;
